pkg/controller/deployment/util/deployment_util_test.go

```diff
@@ -1248,3 +1248,79 @@ func TestGetDeploymentsForReplicaSet(t *testing.T) {
 	}
 
 }
+
+func TestMinAvailable(t *testing.T) {
+	deployment := func(replicas int32, maxUnavailable intstr.IntOrString) *apps.Deployment {
+		return &apps.Deployment{
+			Spec: apps.DeploymentSpec{
+				Replicas: func(i int32) *int32 { return &i }(replicas),
+				Strategy: apps.DeploymentStrategy{
+					RollingUpdate: &apps.RollingUpdateDeployment{
+						MaxSurge:       func(i int) *intstr.IntOrString { x := intstr.FromInt(i); return &x }(int(1)),
+						MaxUnavailable: &maxUnavailable,
+					},
+					Type: apps.RollingUpdateDeploymentStrategyType,
+				},
+			},
+		}
+	}
+	tests := []struct {
+		name       string
+		deployment *apps.Deployment
+		expected   int32
+	}{
+		{
+			name:       "replicas greater than maxUnavailable",
+			deployment: deployment(10, intstr.FromInt(5)),
+			expected:   int32(5),
+		},
+		{
+			name:       "replicas equal maxUnavailable",
+			deployment: deployment(10, intstr.FromInt(10)),
+			expected:   int32(0),
+		},
+		{
+			name:       "replicas less than maxUnavailable",
+			deployment: deployment(5, intstr.FromInt(10)),
+			expected:   int32(0),
+		},
+		{
+			name:       "replicas is 0",
+			deployment: deployment(0, intstr.FromInt(10)),
+			expected:   int32(0),
+		},
+		{
+			name: "minAvailable with Recreate deployment strategy",
+			deployment: &apps.Deployment{
+				Spec: apps.DeploymentSpec{
+					Strategy: apps.DeploymentStrategy{
+						Type: apps.RecreateDeploymentStrategyType,
+					},
+				},
+			},
+			expected: int32(0),
+		},
+		{
+			name:       "replicas greater than maxUnavailable with percents",
+			deployment: deployment(10, intstr.FromString("50%")),
```

Comment:
maybe a different percentage, so that expected is not the same as maxUnavailable.